src/analysis/retail/evoker/preservation/modules/talents/TitansGift.tsx

```diff
@@ -0,0 +1,170 @@
+      this.countBlossomCasts,
+    );
+
+    this.addEventListener(
+      Events.cast.by(SELECTED_PLAYER).spell(TALENTS_EVOKER.ECHO_TALENT),
+      this.countEchoCasts,
+    );
+
+    //Track blossom heals
+    this.addEventListener(
+      Events.heal.by(SELECTED_PLAYER).spell(SPELLS.EMERALD_BLOSSOM),
+      this.emeraldBlossomHeal,
+    );
+
+    //Track echo heals
+    this.addEventListener(
+      Events.heal
+        .by(SELECTED_PLAYER)
+        .spell([
+          SPELLS.DREAM_BREATH_ECHO,
+          SPELLS.EMERALD_BLOSSOM_ECHO,
+          SPELLS.LIVING_FLAME_HEAL,
+          SPELLS.SPIRITBLOOM_SPLIT,
+          SPELLS.SPIRITBLOOM_FONT,
+          SPELLS.SPIRITBLOOM,
+          SPELLS.VERDANT_EMBRACE_HEAL,
+        ]),
+      this.echoHeal,
+    );
+
+    this.addEventListener(
+      Events.applybuff.by(SELECTED_PLAYER).spell([SPELLS.REVERSION_ECHO, SPELLS.DREAM_BREATH_ECHO]),
+      this.echoHeal,
+    );
+
+    this.addEventListener(
+      Events.refreshbuff
+        .by(SELECTED_PLAYER)
+        .spell([SPELLS.REVERSION_ECHO, SPELLS.DREAM_BREATH_ECHO]),
+      this.echoHeal,
+    );
+  }
+
+  //Count total casts
+  countBlossomCasts() {
+    this.totalBlossomsCasted += 1;
+  }
+
+  countEchoCasts() {
+    this.totalEchoesCasted += 1;
+  }
+
+  //Track blossom healing added
+  emeraldBlossomHeal(event: HealEvent) {
+    const BlossomCast = getBlossomCast(event);
+    if (BlossomCast && isCastFromBurst(BlossomCast)) {
+      this.buffedBlossoms += 1;
+      const BlossomHeals = getHealEvents(event);
+      for (const BlossomHeal of BlossomHeals) {
+        this.healingAddedToBlossoms += (BlossomHeal.amount + (BlossomHeal.absorbed ?? 0)) / 5;
+      }
+    }
+  }
+
+  //Track echo healing added
+  echoHeal(event: HealEvent | ApplyBuffEvent | RefreshBuffEvent) {
+    const EchoApplication = getEchoAplication(event);
+    if (EchoApplication && isCastFromBurst(EchoApplication)) {
+      this.buffedEchoes += 1;
+      if (event.type === EventType.Heal) {
+        this.healingAddedToEcho += (event.amount + (event.absorbed ?? 0)) / 5;
+      } else {
+        if (event.ability.name === TALENTS_EVOKER.DREAM_BREATH_TALENT.name) {
+          const DbHealing = getDreamBreathHealing(event);
+          for (const DbHeal of DbHealing) {
+            this.healingAddedToEcho += (DbHeal.amount + (DbHeal.absorbed ?? 0)) / 5;
+          }
+        } else if (event.ability.name === TALENTS_EVOKER.REVERSION_TALENT.name) {
+          const RevHealing = getReversionHealing(event);
+          for (const RevHeal of RevHealing) {
+            this.healingAddedToEcho += (RevHeal.amount + (RevHeal.absorbed ?? 0)) / 5;
+          }
+        }
+      }
+    }
+  }
+
+  statistic() {
+    const PercentBuffedBlossoms =
+      this.totalBlossomsCasted !== 0 ? (this.buffedBlossoms * 100) / this.totalBlossomsCasted : 0;
+    const PercentBuffedEchoes =
+      this.totalEchoesCasted !== 0 ? (this.buffedEchoes * 100) / this.totalEchoesCasted : 0;
```

Comment:
```suggestion
    const percentBuffedBlossoms =
      this.totalBlossomsCasted !== 0 ? (this.buffedBlossoms * 100) / this.totalBlossomsCasted : 0;
    const percentBuffedEchoes =
      this.totalEchoesCasted !== 0 ? (this.buffedEchoes * 100) / this.totalEchoesCasted : 0;
```
nit: use camel casing for local variables. This applies elsewhere but not going to comment in every place.